src/mongoose_riak.erl

```diff
@@ -221,3 +235,44 @@ verify_if_riak_opt_exists(Opt, _) -> Opt.
 maybe_add_additional_opts(Opts, AdditionalOpts) ->
     Opts2 = [verify_if_riak_opt_exists(Opt, []) || Opt <- AdditionalOpts],
     lists:flatten(Opts ++ Opts2).
+
+
+%%%===================================================================
+%%% proxy worker
+%%%===================================================================
+
+%% It is required here because we want to use available_worker strategy, while:
```

Comment:
That's not true. There is function `wpool_pool:next_available_worker` which could be used to select the worker. There is no need to created the intermediate `gen_server`.

Reply:
So the solution would be similar to the one in `mongoose_redis` only with different function used to select the worker.

Reply:
Not exactly - calling wpool with available_worker strategy waits for a worker and times out after a while if nothing becomes available; wpool_pool:next_available_worker immediately returns either a worker or a  no_available_workers atom.

Reply:
And that's perfect, fail fast! We can wait a bit a retry the query if we want.

Reply:
I guess we'd have to, to somehow emulate the behaviour of available_worker. If all workers happen to be busy we don't have to fail, we can wait a few seconds, there is no such rush.
And yes, this might be a way to get rid of this stupid proxy worker. There is one downside, which is granularity - worker pool's queue manager executes a waiting tasks immediately as soon as  worker becomes available, while if we retry every second we wait an additional half a second on average. During which the workers may become busy again.

Having said that, I'm not a fan of the proxy worker stuff. What I'd really like to do is improve worker_pool so that it can be called with MFA or at least FA, not just A.

Reply:
Excellent, let's do it then. I hope Brujo agrees :)

Reply:
The code is ours from what I understand (inaka account on Github is managed by ESL) so we (or some of us) are able to merge a PR as well.